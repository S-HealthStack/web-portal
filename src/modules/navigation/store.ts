import { createSelector } from '@reduxjs/toolkit';
import { matchPath } from 'react-router-dom';
import { createBrowserHistory, createMemoryHistory, Location } from 'history';

import { RootState } from 'src/modules/store/store';

export enum Path {
  Root = '/',

  SignIn = '/signin',
  AccountActivation = '/account/activation',
  AccountCreate = '/account/create',
  AccountConfirm = '/account/confirm',
  AccountVerification = '/email-verification',
  CreateStudy = '/create-study',

  Overview = '/overview',
  OverviewSubject = '/overview/participants/:subjectId',
  TrialManagement = '/trial-management',
  TrialManagementSubject = '/trial-management/participants/:subjectId',
  TrialManagementEditSurvey = '/trial-management/surveys/:surveyId/edit',
  TrialManagementSurveyResults = '/trial-management/surveys/:surveyId/results',
  UserAnalytics = '/user-analytics',
  DataCollection = '/data-collection',
  DataCollectionSubject = '/data-collection/participants/:subjectId',
  StudySettings = '/study-settings',
}

export type LocationState = {
  from: Location;
};

export const makeHistory = () =>
  process.env.NODE_ENV !== 'test'
    ? createBrowserHistory({ basename: process.env.PUBLIC_PATH })
    : createMemoryHistory();

export const history = makeHistory();

const sectionPaths = [
  Path.Overview,
  Path.OverviewSubject,
  Path.TrialManagement,
  Path.UserAnalytics,
  Path.DataCollection,
  Path.StudySettings,
] as const;

export const sectionPathSelector = createSelector(
  (state: RootState) => state.router,
  (r) => sectionPaths.find((s) => matchPath((r.location as unknown as Location).pathname, s))
);
