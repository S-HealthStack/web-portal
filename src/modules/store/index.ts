export type { RootState, AppDispatch, AppThunk } from './store';
export { useAppDispatch, useAppSelector } from './hooks';
export * from './tools';
