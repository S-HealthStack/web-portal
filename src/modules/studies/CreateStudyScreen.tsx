import React, { useCallback, useEffect, useState } from 'react';
import CreateStudyCard from 'src/modules/studies/CreateStudyCard';
import SelectRoleCard from 'src/modules/studies/SelectRoleCard';
import styled from 'styled-components';

import { SpecColorType } from 'src/styles/theme';
import { px } from 'src/styles';
import StudyLayout from 'src/modules/studies/StudyLayout';

const MainWrapper = styled.div`
  display: flex;
  height: 100%;
  width: 100%;
  @media (max-height: ${px(475)}) {
    overflow-y: scroll;
  }
  @media (max-width: ${px(665)}) {
    overflow-x: scroll;
  }
`;

const CreateStudyScreen: React.FC = () => {
  const [card, setCard] = useState<'createStudy' | 'selectRole'>('createStudy');
  const [name, setName] = useState<string>('');
  const [color, setColor] = useState<SpecColorType | undefined>(undefined);

  const handleSetStudyParams = useCallback((studyName: string, avatarColor: SpecColorType) => {
    setName(studyName);
    setColor(avatarColor);
  }, []);

  useEffect(() => {
    if (name && color && card === 'createStudy') {
      setCard('selectRole');
    }
  }, [name, color, card]);

  return (
    <StudyLayout>
      <MainWrapper data-testid="create-study">
        {card === 'createStudy' && <CreateStudyCard onClick={handleSetStudyParams} />}
        {card === 'selectRole' && <SelectRoleCard color={color || 'primary'} name={name} />}
      </MainWrapper>
    </StudyLayout>
  );
};

export default CreateStudyScreen;
