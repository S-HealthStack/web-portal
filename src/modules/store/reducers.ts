import { combineReducers } from '@reduxjs/toolkit';
import { connectRouter } from 'connected-react-router';
import type { Reducer } from 'redux';

import authReducer from 'src/modules/auth/auth.slice';
import studySettingsReducer from 'src/modules/study-settings/studySettings.slice';
import snackbarReducer from 'src/modules/snackbar/snackbar.slice';
import overviewReducers from 'src/modules/overview/overview.slice';
import studiesReducer from 'src/modules/studies/studies.slice';
import trialManagementReducer from 'src/modules/trial-management/trialManagement.slice';
import dataCollectionReducer from 'src/modules/data-collection/dataCollection.slice';
import sidebarReducer from 'src/modules/main-layout/sidebar/sidebar.slice';
import { history } from 'src/modules/navigation/store';
import { DataSliceState } from 'src/modules/store/createDataSlice';

const createReducer = <T, V>(asyncReducers?: { [key: string]: Reducer<DataSliceState<T, V>> }) =>
  combineReducers({
    ...authReducer,
    router: connectRouter(history),
    snackbar: snackbarReducer,
    studies: studiesReducer,
    dataCollection: dataCollectionReducer,
    sidebar: sidebarReducer,
    ...studySettingsReducer,
    ...overviewReducers,
    ...trialManagementReducer,
    ...asyncReducers,
  });

export default createReducer;
