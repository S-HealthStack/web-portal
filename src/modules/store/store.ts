/* eslint-disable import/no-import-module-exports */
import {
  Action,
  ThunkAction,
  configureStore,
  getDefaultMiddleware,
  ThunkDispatch,
} from '@reduxjs/toolkit';
import { routerMiddleware } from 'connected-react-router';

import { history } from 'src/modules/navigation/store';
import rootReducer from 'src/modules/store/reducers';

export { rootReducer };

export const store = configureStore({
  reducer: rootReducer,
  middleware: [...getDefaultMiddleware(), routerMiddleware(history)],
  devTools: process.env.NODE_ENV !== 'production',
});

if (module.hot) {
  // Enable Webpack hot module replacement for reducers
  module.hot.accept('./reducers', async () => {
    // eslint-disable-next-line import/no-useless-path-segments
    const nextRootReducer = (await import('../store/reducers')).default;
    store.replaceReducer(nextRootReducer);
  });
}

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = ThunkDispatch<RootState, unknown, Action>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
