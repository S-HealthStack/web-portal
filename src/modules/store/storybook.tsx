import React from 'react';
import { configureStore, ConfigureStoreOptions } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { DecoratorFunction } from '@storybook/csf';
import { ReactFramework } from '@storybook/react';
import { MemoryRouter } from 'react-router-dom';
import { rootReducer, RootState } from './store';

export const getStoreDecorator = (state: ConfigureStoreOptions<RootState>['preloadedState']) => {
  const store = configureStore({
    reducer: rootReducer,
    preloadedState: state,
  });

  return ((story) => (
    <Provider store={store}>
      <MemoryRouter>{story()}</MemoryRouter>
    </Provider>
  )) as DecoratorFunction<ReactFramework>;
};
