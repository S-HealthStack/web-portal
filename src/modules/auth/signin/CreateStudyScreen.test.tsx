import React from 'react';
import { act, render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import { ConnectedRouter, getLocation } from 'connected-react-router';
import { matchPath } from 'react-router-dom';

import 'jest-styled-components';
import 'src/__mocks__/setupWindowMatchMediaMock';
import 'src/__mocks__/setupResizeObserverMock';

import theme from 'src/styles/theme';
import { makeStore } from 'src/modules/store/store';
import { makeHistory, Path } from 'src/modules/navigation/store';
import CreateStudyScreen, { avatarColors } from 'src/modules/studies/CreateStudyScreen';

describe('CreateStudyScreen', () => {
  let store: ReturnType<typeof makeStore>;
  let history: ReturnType<typeof makeHistory>;

  beforeEach(() => {
    history = makeHistory();
    store = makeStore(history);
  });

  it('should create study', async () => {
    await act(() => {
      render(
        <ThemeProvider theme={theme}>
          <Provider store={store}>
            <ConnectedRouter history={history}>
              <CreateStudyScreen />
            </ConnectedRouter>
          </Provider>
        </ThemeProvider>
      );
    });

    expect(await screen.findByTestId('create-study')).toMatchSnapshot();

    const name = await screen.findByTestId('create-study-name');
    const avatars = await screen.findAllByTestId('create-study-avatar');
    const firstAvatar = avatars[0];
    const send = await screen.findByTestId('create-study-send');

    expect(avatars).toHaveLength(avatarColors.length);

    const nameValue = 'title';
    await userEvent.type(name, nameValue);
    expect(name).toHaveValue(nameValue);

    await userEvent.hover(firstAvatar);
    await userEvent.click(firstAvatar);
    await userEvent.unhover(firstAvatar);

    await waitFor(() => expect(send).not.toHaveAttribute('disabled'));

    await userEvent.click(send);

    await waitFor(() =>
      expect(
        matchPath(getLocation(store.getState()).pathname, {
          path: Path.StudySettings,
          exact: true,
        })
      ).not.toBeNull()
    );
  });

  it('[NEGATIVE] should create study with partial data', async () => {
    await act(() => {
      render(
        <ThemeProvider theme={theme}>
          <Provider store={store}>
            <ConnectedRouter history={history}>
              <CreateStudyScreen />
            </ConnectedRouter>
          </Provider>
        </ThemeProvider>
      );
    });

    expect(await screen.findByTestId('create-study')).toMatchSnapshot();

    const name = await screen.findByTestId('create-study-name');
    const avatars = await screen.findAllByTestId('create-study-avatar');
    const send = await screen.findByTestId('create-study-send');

    expect(avatars).toHaveLength(avatarColors.length);

    const nameValue = 'title';
    await userEvent.type(name, nameValue);
    expect(name).toHaveValue(nameValue);

    expect(send).toHaveAttribute('disabled');

    await userEvent.click(send);

    await waitFor(() =>
      expect(
        matchPath(getLocation(store.getState()).pathname, {
          path: Path.StudySettings,
          exact: true,
        })
      ).toBeNull()
    );
  });
});
