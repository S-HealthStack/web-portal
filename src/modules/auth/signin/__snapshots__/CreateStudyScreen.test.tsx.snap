// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateStudyScreen [NEGATIVE] should create study with partial data 1`] = `
.c19 {
  position: absolute;
  will-change: opacity,transform;
  border-radius: 50%;
  -webkit-animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  z-index: 0;
  pointer-events: none;
}

.c17 {
  position: relative;
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.25rem;
  width: 100%;
  white-space: nowrap;
  padding: 0;
  overflow: hidden;
  -webkit-transition-duration: 400ms;
  transition-duration: 400ms;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  height: 3rem;
  border: none;
  background-color: #4475E3;
}

.c17 > div {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 600;
  font-style: normal;
}

.c17 > div svg {
  margin-right: 0.25rem;
}

.c17 > div {
  color: #FFFFFF;
}

.c17 > div svg {
  fill: #FFFFFF;
}

.c17:hover:enabled {
  background-color: #2F65E0;
}

.c17:disabled {
  background-color: #B3C6F1;
}

.c17:focus-visible:enabled {
  border: 0.125rem solid #FFFFFF;
  background-color: #4475E3;
  outline: 0.125rem solid #C7D6F7;
}

.c18 {
  position: relative;
  z-index: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transform: translate3d(0,0,0.0625rem);
  -ms-transform: translate3d(0,0,0.0625rem);
  transform: translate3d(0,0,0.0625rem);
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
}

.c9 > svg {
  position: relative;
  bottom: 2.6875rem;
  left: calc(100% - 0.5rem);
}

.c8 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  box-sizing: border-box;
  height: 3.5rem;
  width: 100%;
  margin: 0;
  padding: 1rem;
  background-color: #F7F8FA;
  border: 0.0625rem solid #F7F8FA;
  border-radius: 0.25rem;
  -webkit-transition: border 300ms cubic-bezier(0.45,0.05,0,1);
  transition: border 300ms cubic-bezier(0.45,0.05,0,1);
  caret-color: #4475E3;
}

.c8:hover:enabled {
  border-color: #2F65E0;
}

.c8:active:enabled,
.c8:focus-visible {
  outline: none;
  border-color: #4475E3;
}

.c8:disabled {
  color: #DBDBDB;
  border-color: rgba(219,219,219,0.2);
  background-color: rgba(219,219,219,0.2);
}

.c8::-webkit-input-placeholder {
  color: #808080;
}

.c8::-moz-placeholder {
  color: #808080;
}

.c8:-ms-input-placeholder {
  color: #808080;
}

.c8::placeholder {
  color: #808080;
}

.c8:disabled::-webkit-input-placeholder {
  color: #DBDBDB;
}

.c8:disabled::-moz-placeholder {
  color: #DBDBDB;
}

.c8:disabled:-ms-input-placeholder {
  color: #DBDBDB;
}

.c8:disabled::placeholder {
  color: #DBDBDB;
}

.c8:-webkit-autofill,
.c8:-webkit-autofill:hover,
.c8:-webkit-autofill:focus {
  border: 0.0625rem solid #4475E3;
  -webkit-transition: background-color 5000s ease-in-out 0s;
  transition: background-color 5000s ease-in-out 0s;
  background-color: #F7F8FA !important;
  background-image: unset !important;
  -webkit-text-fill-color: #838383;
}

.c8:-webkit-autofill:focus {
  border-color: #4475E3;
}

.c6 {
  max-height: 3.5rem;
}

.c6:hover .c7:enabled {
  border-color: #2F65E0;
}

.c5 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  gap: 0.5rem;
  height: 1.125rem;
}

.c10 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #D14343;
  padding-left: 1rem;
  height: 1.0625rem;
}

.c16 {
  border-radius: 1.5625rem;
  height: 3.875rem;
  -webkit-transition: opacity 300ms cubic-bezier(0.45,0.05,0,1);
  transition: opacity 300ms cubic-bezier(0.45,0.05,0,1);
  position: relative;
  background-color: transparent !important;
  opacity: 1;
}

.c16:hover {
  opacity: 1;
  box-shadow: 0 0.9375rem 1.25rem rgba(180,180,180,0.3);
  cursor: pointer;
}

.c16:hover:active {
  opacity: 1;
  box-shadow: 0 0.9375rem 1.25rem rgba(180,180,180,0.3);
  cursor: pointer;
  -webkit-filter: drop-shadow(0 0.9375rem 1.25rem rgba(180,180,180,0.3));
  filter: drop-shadow(0 0.9375rem 1.25rem rgba(180,180,180,0.3));
}

.c1 {
  color: #474747;
  min-width: 41.625rem;
  width: 55.556vw;
  min-height: 29.761875rem;
  height: 39.722539999999995vw;
  max-height: 90vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: auto;
  background-color: #FFFFFF;
  box-shadow: 0 0 0.125rem 0rem rgba(0,0,0,0.15);
  border-radius: 0.25rem;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
}

.c2 {
  height: 25.75rem;
  width: 28rem;
  margin: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  font-family: Inter,sans-serif;
  font-size: 2rem;
  line-height: 2.625rem;
  font-weight: 600;
  font-style: normal;
  margin: 0 auto 2.5rem;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  height: 7.9375rem;
  width: 100%;
  margin: 0.5rem 0 2.625rem;
}

.c12 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  height: 2.625rem;
}

.c13 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #808080;
  line-height: 1.5625rem;
}

.c14 {
  position: relative;
  width: 100%;
  padding-top: 0.125rem;
  height: 4.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c15 {
  width: 4.5rem;
  height: 4.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding-top: 0.25rem;
}

@media (max-height:29.6875rem) {
  .c0 {
    overflow-y: scroll;
  }
}

@media (max-width:41.5625rem) {
  .c0 {
    overflow-x: scroll;
  }
}

<div
  class="c0"
  data-testid="create-study"
>
  <div
    class="c1"
    width="55.556"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        Create a study
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
          data-testid="input-description"
        >
          Study Name
        </div>
        <div
          class="c6"
        >
          <input
            class="c7 c8"
            data-testid="create-study-name"
            placeholder="Name your study"
            tabindex="0"
            type="text"
            value=""
          />
          <div
            class="c9"
          />
        </div>
        <div
          class="c10"
          data-testid="input-error"
        >
          Â 
        </div>
      </div>
      <div
        class="c11"
      >
        <div
          class="c12"
        >
          Study Logo
          <div
            class="c13"
          >
            Select a logo to represent your study
          </div>
        </div>
        <div
          class="c14"
          tabindex="0"
        >
          <div
            class="c15"
          >
            <div
              class="c16"
              color="secondarySkyblue"
              data-testid="create-study-avatar"
            >
              <svg
                data-testid="avatar-icon"
                fill="#00B0D7"
              />
            </div>
          </div>
          <div
            class="c15"
          >
            <div
              class="c16"
              color="secondaryViolet"
              data-testid="create-study-avatar"
            >
              <svg
                data-testid="avatar-icon"
                fill="#944ED7"
              />
            </div>
          </div>
          <div
            class="c15"
          >
            <div
              class="c16"
              color="secondaryTangerine"
              data-testid="create-study-avatar"
            >
              <svg
                data-testid="avatar-icon"
                fill="#F39C18"
              />
            </div>
          </div>
          <div
            class="c15"
          >
            <div
              class="c16"
              color="secondaryGreen"
              data-testid="create-study-avatar"
            >
              <svg
                data-testid="avatar-icon"
                fill="#1AD598"
              />
            </div>
          </div>
          <div
            class="c15"
          >
            <div
              class="c16"
              color="secondaryRed"
              data-testid="create-study-avatar"
            >
              <svg
                data-testid="avatar-icon"
                fill="#FA5F4F"
              />
            </div>
          </div>
        </div>
      </div>
      <button
        class="c17"
        data-testid="create-study-send"
        disabled=""
        fill="solid"
        style="overflow: hidden; position: relative;"
      >
        <div
          class="c18"
          data-testid="content"
        >
          Create Study
        </div>
        <div
          class="c19"
          color="primaryPressed"
          opacity="1,1"
          scale="0"
          style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`CreateStudyScreen should create study 1`] = `
.c19 {
  position: absolute;
  will-change: opacity,transform;
  border-radius: 50%;
  -webkit-animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  z-index: 0;
  pointer-events: none;
}

.c17 {
  position: relative;
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.25rem;
  width: 100%;
  white-space: nowrap;
  padding: 0;
  overflow: hidden;
  -webkit-transition-duration: 400ms;
  transition-duration: 400ms;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  height: 3rem;
  border: none;
  background-color: #4475E3;
}

.c17 > div {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 600;
  font-style: normal;
}

.c17 > div svg {
  margin-right: 0.25rem;
}

.c17 > div {
  color: #FFFFFF;
}

.c17 > div svg {
  fill: #FFFFFF;
}

.c17:hover:enabled {
  background-color: #2F65E0;
}

.c17:disabled {
  background-color: #B3C6F1;
}

.c17:focus-visible:enabled {
  border: 0.125rem solid #FFFFFF;
  background-color: #4475E3;
  outline: 0.125rem solid #C7D6F7;
}

.c18 {
  position: relative;
  z-index: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transform: translate3d(0,0,0.0625rem);
  -ms-transform: translate3d(0,0,0.0625rem);
  transform: translate3d(0,0,0.0625rem);
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
}

.c9 > svg {
  position: relative;
  bottom: 2.6875rem;
  left: calc(100% - 0.5rem);
}

.c8 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  box-sizing: border-box;
  height: 3.5rem;
  width: 100%;
  margin: 0;
  padding: 1rem;
  background-color: #F7F8FA;
  border: 0.0625rem solid #F7F8FA;
  border-radius: 0.25rem;
  -webkit-transition: border 300ms cubic-bezier(0.45,0.05,0,1);
  transition: border 300ms cubic-bezier(0.45,0.05,0,1);
  caret-color: #4475E3;
}

.c8:hover:enabled {
  border-color: #2F65E0;
}

.c8:active:enabled,
.c8:focus-visible {
  outline: none;
  border-color: #4475E3;
}

.c8:disabled {
  color: #DBDBDB;
  border-color: rgba(219,219,219,0.2);
  background-color: rgba(219,219,219,0.2);
}

.c8::-webkit-input-placeholder {
  color: #808080;
}

.c8::-moz-placeholder {
  color: #808080;
}

.c8:-ms-input-placeholder {
  color: #808080;
}

.c8::placeholder {
  color: #808080;
}

.c8:disabled::-webkit-input-placeholder {
  color: #DBDBDB;
}

.c8:disabled::-moz-placeholder {
  color: #DBDBDB;
}

.c8:disabled:-ms-input-placeholder {
  color: #DBDBDB;
}

.c8:disabled::placeholder {
  color: #DBDBDB;
}

.c8:-webkit-autofill,
.c8:-webkit-autofill:hover,
.c8:-webkit-autofill:focus {
  border: 0.0625rem solid #4475E3;
  -webkit-transition: background-color 5000s ease-in-out 0s;
  transition: background-color 5000s ease-in-out 0s;
  background-color: #F7F8FA !important;
  background-image: unset !important;
  -webkit-text-fill-color: #838383;
}

.c8:-webkit-autofill:focus {
  border-color: #4475E3;
}

.c6 {
  max-height: 3.5rem;
}

.c6:hover .c7:enabled {
  border-color: #2F65E0;
}

.c5 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  gap: 0.5rem;
  height: 1.125rem;
}

.c10 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #D14343;
  padding-left: 1rem;
  height: 1.0625rem;
}

.c16 {
  border-radius: 1.5625rem;
  height: 3.875rem;
  -webkit-transition: opacity 300ms cubic-bezier(0.45,0.05,0,1);
  transition: opacity 300ms cubic-bezier(0.45,0.05,0,1);
  position: relative;
  background-color: transparent !important;
  opacity: 1;
}

.c16:hover {
  opacity: 1;
  box-shadow: 0 0.9375rem 1.25rem rgba(180,180,180,0.3);
  cursor: pointer;
}

.c16:hover:active {
  opacity: 1;
  box-shadow: 0 0.9375rem 1.25rem rgba(180,180,180,0.3);
  cursor: pointer;
  -webkit-filter: drop-shadow(0 0.9375rem 1.25rem rgba(180,180,180,0.3));
  filter: drop-shadow(0 0.9375rem 1.25rem rgba(180,180,180,0.3));
}

.c1 {
  color: #474747;
  min-width: 41.625rem;
  width: 55.556vw;
  min-height: 29.761875rem;
  height: 39.722539999999995vw;
  max-height: 90vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: auto;
  background-color: #FFFFFF;
  box-shadow: 0 0 0.125rem 0rem rgba(0,0,0,0.15);
  border-radius: 0.25rem;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
}

.c2 {
  height: 25.75rem;
  width: 28rem;
  margin: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  font-family: Inter,sans-serif;
  font-size: 2rem;
  line-height: 2.625rem;
  font-weight: 600;
  font-style: normal;
  margin: 0 auto 2.5rem;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  height: 7.9375rem;
  width: 100%;
  margin: 0.5rem 0 2.625rem;
}

.c12 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  height: 2.625rem;
}

.c13 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #808080;
  line-height: 1.5625rem;
}

.c14 {
  position: relative;
  width: 100%;
  padding-top: 0.125rem;
  height: 4.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c15 {
  width: 4.5rem;
  height: 4.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding-top: 0.25rem;
}

@media (max-height:29.6875rem) {
  .c0 {
    overflow-y: scroll;
  }
}

@media (max-width:41.5625rem) {
  .c0 {
    overflow-x: scroll;
  }
}

<div
  class="c0"
  data-testid="create-study"
>
  <div
    class="c1"
    width="55.556"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        Create a study
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
          data-testid="input-description"
        >
          Study Name
        </div>
        <div
          class="c6"
        >
          <input
            class="c7 c8"
            data-testid="create-study-name"
            placeholder="Name your study"
            tabindex="0"
            type="text"
            value=""
          />
          <div
            class="c9"
          />
        </div>
        <div
          class="c10"
          data-testid="input-error"
        >
          Â 
        </div>
      </div>
      <div
        class="c11"
      >
        <div
          class="c12"
        >
          Study Logo
          <div
            class="c13"
          >
            Select a logo to represent your study
          </div>
        </div>
        <div
          class="c14"
          tabindex="0"
        >
          <div
            class="c15"
          >
            <div
              class="c16"
              color="secondarySkyblue"
              data-testid="create-study-avatar"
            >
              <svg
                data-testid="avatar-icon"
                fill="#00B0D7"
              />
            </div>
          </div>
          <div
            class="c15"
          >
            <div
              class="c16"
              color="secondaryViolet"
              data-testid="create-study-avatar"
            >
              <svg
                data-testid="avatar-icon"
                fill="#944ED7"
              />
            </div>
          </div>
          <div
            class="c15"
          >
            <div
              class="c16"
              color="secondaryTangerine"
              data-testid="create-study-avatar"
            >
              <svg
                data-testid="avatar-icon"
                fill="#F39C18"
              />
            </div>
          </div>
          <div
            class="c15"
          >
            <div
              class="c16"
              color="secondaryGreen"
              data-testid="create-study-avatar"
            >
              <svg
                data-testid="avatar-icon"
                fill="#1AD598"
              />
            </div>
          </div>
          <div
            class="c15"
          >
            <div
              class="c16"
              color="secondaryRed"
              data-testid="create-study-avatar"
            >
              <svg
                data-testid="avatar-icon"
                fill="#FA5F4F"
              />
            </div>
          </div>
        </div>
      </div>
      <button
        class="c17"
        data-testid="create-study-send"
        disabled=""
        fill="solid"
        style="overflow: hidden; position: relative;"
      >
        <div
          class="c18"
          data-testid="content"
        >
          Create Study
        </div>
        <div
          class="c19"
          color="primaryPressed"
          opacity="1,1"
          scale="0"
          style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
        />
      </button>
    </div>
  </div>
</div>
`;
